int32 seed
---
bool success 
